src/backend/distributed/deparser/ruleutils_12.c

```diff
@@ -5181,20 +5181,37 @@ get_rule_expr(Node *node, deparse_context *context,
 		case T_RelabelType:
 			{
 				RelabelType *relabel = (RelabelType *) node;
-				Node	   *arg = (Node *) relabel->arg;
 
-				if (relabel->relabelformat == COERCE_IMPLICIT_CAST &&
-					!showimplicit)
+				if (relabel->resultcollid != InvalidOid)
```

Comment:
can you add a comment that this is a Citus-specific modification? That will make adding support for future PG releases slightly easier.